pkg/snapshot/snapshot_test.go

```diff
@@ -391,6 +391,160 @@ func TestSnasphotPreservesFileOrder(t *testing.T) {
 	}
 }
 
+func TestSnasphotPreservesWhiteoutOrder(t *testing.T) {
+	newFiles := map[string]string{
+		"foo":     "newbaz1",
+		"bar/bat": "baz",
+		"bar/qux": "quuz",
+		"qux":     "quuz",
+		"corge":   "grault",
+		"garply":  "waldo",
+		"fred":    "plugh",
+		"xyzzy":   "thud",
+	}
+
+	newFileNames := []string{}
+
+	for fileName := range newFiles {
+		newFileNames = append(newFileNames, fileName)
+	}
+
+	filesInTars := [][]string{}
+
+	for i := 0; i <= 2; i++ {
+		testDir, snapshotter, cleanup, err := setUpTest()
```

Comment:
The code is not ideal (using defer in a loop) and the non-determinism is a little hard to understand (converting from a map to a slice). But I chose to impolement the new tests like `TestSnasphotPreservesFileOrder` for consistency.